### PR TITLE
[EOS-S3] Changes to common CMake and Python scripts

### PR DESCRIPTION
**Pull request by mkurc-ant**
This PR incorporates all changes needed for common CMake scripts as well as for common utility Python scripts required for QuickLogic EOS-S3 eFPGA support.

### CODE REVIEW

**Comment by acomodi** [comment]
Might be worth squashing all commits

**Comment by mkurc-ant** [comment]
@acomodi I did some squashing / cleanup but wanted to preserve as much of the history as possible

**Comment by acomodi** [comment]
@mkurc-ant just a nitpick actually, but I think that it would be best to just squash everything in one/two commits.

Take this for example: https://github.com/SymbiFlow/symbiflow-arch-defs/pull/2149/commits/97f547f83ad84659a46b16455b268e7b71d2b65c. It just removes one line and does specify that something else is done in the commit description.

I think that it would make more sense to squash everything in a couple of commits, and maintaining the co-authors, as well as the various commit messages.

**Comment by mkurc-ant** [comment]
@acomodi Ok, Now there is only one comment

File: utils/vpr_io_place.py [hunk 1]
```diff
@@ -22,6 +22,7 @@ def __init__(self):
         self.inputs = set()
         self.outputs = set()
         self.net_to_block = None
+        self.block_to_inst = {}
```
Comment (acomodi) [inline utils/vpr_io_place.py:25]: Just saw this, probably it is redundant as it is already defined below.

Comment (mkurc-ant) [reply utils/vpr_io_place.py]: Right... fixed.

**Review by acomodi — APPROVED** [review]
LGTM

**Pull request merged**